### ISSUE TITLE
qml wizard: password not validated during wallet creation

### ISSUE THREAD

**Issue by SomberNight** [spesmilo/electrum#8280]
Looks like this line is ignored:
https://github.com/spesmilo/electrum/blob/965e1ac9a3d8c8741b6f7a46bf4e4da91dd86aa1/electrum/gui/qml/components/wizard/WCWalletPassword.qml#L8

I can create a wallet without a password, or with a shorter than 6 char password, and equality between the two text fields is not checked either.

**Comment by accumulator** [comment]
> Looking at Wizard.qml, I guess this might have something to do with the password screen being the "last" screen, as in other screens the `valid` attribute is checked properly.

No, the cause is a delayed valid check on the seed entry page, triggered by a spurious textChanged event when moving to the next wizard page, overriding the valid state of the wizard as this event comes later than the (immediate) state of the last page.

As these textChanged events should not occur since the text value actually doesn't change, it looks like a Qt bug.

**Comment by ecdsa** [comment]
This seems hard to debug, and I am having a hard time understanding the workaround.
Isn't the problem that you allow `pages.pagevalid` to be set by multiple pages simultaneously?
How about checkout that only the current page is allowed to set that field?

The following fix seems to work:

```diff --git a/electrum/gui/qml/components/wizard/Wizard.qml b/electrum/gui/qml/components/wizard/Wizard.qml
index c2c38329e..918bc962f 100644
--- a/electrum/gui/qml/components/wizard/Wizard.qml
+++ b/electrum/gui/qml/components/wizard/Wizard.qml
@@ -63,6 +63,7 @@ ElDialog {
         Object.assign(wdata_copy, wdata)
         var page = comp.createObject(pages, {wizard_data: wdata_copy})
         page.validChanged.connect(function() {
+            if(page != pages.contentChildren[pages.currentIndex]) return
             pages.pagevalid = page.valid
         } )
         page.lastChanged.connect(function() {
```

**Comment by accumulator** [comment]
Yes, this should be added too

**Comment by ecdsa** [comment]
I do not understand, is 78d7929 still necessary? Can you  explain what it does? 
I understand that the issue is spurious events, but I do not see how that commit interacts with them.

**Comment by accumulator** [comment]
I think we should keep 78d79290ad156122075d38c7ba7786d13821d2fd in, to make sure `SeedTextArea` as a component behaves as expected.

The `SeedTextArea.text` property previously was defined as an alias into the `TextArea.text` property. Apparently `TextArea.text` somehow behaves differently than usual (either `TextArea` does something strange, or the property alias triggers this behaviour). This is now replaced by a 'normal' property, which I propagate to and from `TextArea.text`. 

I think we can even remove the `if (root.text != seedtextarea.text)` guard as the property change event should only trigger when it actually differs.

**Comment by accumulator** [comment]
Note: Qt bindings can be quirky at times. See for example the fix for seemingly strange issue #8262 is fixed by changing a property reference in bindings to a intermediate local property.
